src/main/java/com/labsynch/labseer/api/ApiSaltController.java

```diff
@@ -95,21 +293,63 @@ public ResponseEntity<String> listJson() {
 		return new ResponseEntity<String>(Salt.toJsonArray(Salt.findAllSalts()), headers, HttpStatus.OK);
 	}
 
+	@RequestMapping(value = "/sdf", method = RequestMethod.GET, headers = "Accept=application/json")
+	@ResponseBody
+	public ResponseEntity<String> listSDF() {
+		HttpHeaders headers = new HttpHeaders();
+		headers.add("Content-Type", "application/text; charset=utf-8");
+		headers.add("Access-Control-Allow-Origin", "*");
+		headers.add("Access-Control-Allow-Headers", "Content-Type");
+		headers.add("Cache-Control", "no-store, no-cache, must-revalidate"); // HTTP 1.1
+		headers.add("Pragma", "no-cache"); // HTTP 1.0
+		headers.setExpires(0); // Expire the cache
+		String SaltSDFStr = saltService.exportSalts(); // Returns SDF String of All Salts Stored
+		return new ResponseEntity<String>(SaltSDFStr, headers, HttpStatus.OK);
+	}
+
 	@RequestMapping(method = RequestMethod.POST, headers = "Accept=application/json")
 	public ResponseEntity<String> createFromJson(@RequestBody String json) throws CmpdRegMolFormatException {
 		HttpHeaders headers = new HttpHeaders();
 		headers.add("Content-Type", "application/text");
 		headers.add("Access-Control-Allow-Origin", "*");
 		headers.add("Access-Control-Allow-Headers", "Content-Type");
-		Salt salt = Salt.fromJsonToSalt(json);
+		headers.add("Cache-Control", "no-store, no-cache, must-revalidate"); // HTTP 1.1
+		headers.add("Pragma", "no-cache"); // HTTP 1.0
+		headers.setExpires(0); // Expire the cache
+
+		Salt salt = null; 
+		boolean dryrun = true; 
+		try
+		{
+			ObjectMapper objectMapper = new ObjectMapper();
+
+			JsonNode rootNode = objectMapper.readTree(json);
+			JsonNode dryRunNode = rootNode.path("dryrun");
```

Comment:
Similar thing here - I think it'd make sense to put the `dryrun` as a RequestParameter